skrub/tests/test_drop_nulls.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from skrub import TableVectorizer
+from skrub import _dataframe as sbd
+from skrub._drop_null import DropNullColumn
+
+
+@pytest.fixture
+def drop_null_table(df_module):
+    return df_module.make_dataframe(
+        {
+            "idx": [
+                1,
+                2,
+                3,
+            ],
+            "value_nan": [
+                np.nan,
+                np.nan,
+                np.nan,
+            ],
+            "value_null": [
+                None,
+                None,
+                None,
+            ],
+            "value_almost_nan": [
+                2.5,
+                np.nan,
+                np.nan,
+            ],
+            "value_almost_null": [
+                "almost",
+                None,
+                None,
+            ],
+        }
+    )
+
+
+def test_single_column(drop_null_table):
+    """Check that null columns are dropped and non-null columns are kept."""
+    dn = DropNullColumn()
+    assert dn.fit_transform(drop_null_table["value_nan"]) == []
+    assert dn.fit_transform(drop_null_table["value_null"]) == []
+
+    assert_array_equal(
+        sbd.to_numpy(sbd.col(drop_null_table, "idx")), np.array([1, 2, 3])
+    )
+
+    assert_array_equal(
+        sbd.to_numpy(sbd.col(drop_null_table, "value_almost_nan")),
+        np.array([2.5, np.nan, np.nan]),
+    )
+
+    # assert_array_equal(
+    #     sbd.to_numpy(sbd.col(drop_null_table, "value_almost_null")),
+    #     np.array(["almost", None, None]),
```

Comment:
Not sure how to write this check so that it works with either pandas or polars

Reply:
You could use `df_module` as a fixture in the test by adding it to the arguments, then comparing series instead of numpy arrays:

```
df_module.assert_column_equal(
    sbd.col(drop_null_table, "value_almost_null"),
    df_module.make_column("value_almost_null", ["almost", None, None]),
)
```

Reply:
This also circumvents that depending on the version of pandas, null values are not treated the same